storage/google_cloud/google_cloud_utils.py

```diff
@@ -84,9 +86,30 @@ def upload_file_to_blob(bucket_credentials_file_path, target_blob_url, f):
     :type target_blob_url: str
     :param f: File to upload, opened in binary mode.
     :type f: file-like
+    :param max_retries: Maximum number of times to retry uploading the file.
+    :type max_retries: int
+    :param blob_chunk_size: The chunk size to use for resumable uploads, in KiB.
+    :type blob_chunk_size: float
     """
-    log.info(f"Uploading file to blob '{target_blob_url}'...")
-    storage_client = storage.Client.from_service_account_json(bucket_credentials_file_path)
-    blob = _blob_at_url(storage_client, target_blob_url)
-    blob.upload_from_file(f)
-    log.info(f"Uploaded file to blob")
+    try:
+        log.info(f"Uploading file to blob '{target_blob_url}'...")
+        storage_client = storage.Client.from_service_account_json(bucket_credentials_file_path)
+        blob = _blob_at_url(storage_client, target_blob_url)
+        blob.chunk_size = blob_chunk_size * 1024
+        blob.upload_from_file(f)
+        log.info(f"Uploaded file to blob")
+
+    except (ConnectionError, socket.timeout, Timeout) as ex:
+        log.warning("Failed to upload due to connection/timeout error")
+        if max_retries > 0 and blob_chunk_size > 256:
```

Comment:
I think this would be cleaner if you handled the error then normal case separately. So:
```
if max_retries <= 0:
    log.error("...")
    raise ex

if blob_chunk_size < 256:
    log.error("...")
    raise ex

log.info("Retrying...")
upload_file_to_blob
```

Reply:
updated thank!